### PR TITLE
Pricelist - QA change : handling inactive records

### PR DESCRIPTION
**Pull request by kumarp-v**

### CODE REVIEW

File: VIS/Areas/VIS/Models/Callouts/MPriceListVersionModel.cs [hunk 1]
```diff
@@ -142,62 +153,91 @@ private void SetTableAndColumnName(ScreenType screenType)
         /// <returns>Price List Version Id</returns>
         public int GetM_PriceList_Version_ID(Ctx ctx, string priceListId, string transactionId, string productId, string uomId, string attrSetInstId)
         {
-            if (Util.GetValueOfInt(transactionId) > 0)
```
Comment (Manjot-Vienna) [inline VIS/Areas/VIS/Models/Callouts/MPriceListVersionModel.cs:145]: Commenting is missing

File: VIS/Areas/VIS/Models/Callouts/MPriceListVersionModel.cs [hunk 2]
```diff
@@ -142,62 +153,91 @@ private void SetTableAndColumnName(ScreenType screenType)
...
-                    {                      
-                        sql = "SELECT plv.M_PriceList_Version_ID FROM M_PriceList_Version plv "
-                                + " JOIN M_ProductPrice pp ON plv.M_PriceList_Version_ID = pp.M_PriceList_Version_ID "
-                                + " WHERE plv.IsActive = 'Y' AND plv.M_PriceList_ID = " + priceListId
-                                + @" AND plv.VALIDFROM <= (SELECT " + _columnName + " FROM " + _tableName + " WHERE " + _keyColumnName + "=" + Util.GetValueOfInt(transactionId) + ") "
-                                + " AND NVL(pp.M_Product_ID, 0) = " + Util.GetValueOfInt(productId)                                
-                                + " AND NVL(pp.C_UOM_ID, 0) = " + Util.GetValueOfInt(uomId)
-                                + " ORDER BY plv.VALIDFROM DESC, plv.M_PriceList_Version_ID DESC";
+                    if (Util.GetValueOfInt(productId) > 0)
+                    {
```
Comment (Manjot-Vienna) [inline VIS/Areas/VIS/Models/Callouts/MPriceListVersionModel.cs:165]: Commenting is missing

File: VIS/Areas/VIS/Models/Callouts/MProductPricingModel.cs [hunk 3]
```diff
@@ -56,6 +57,19 @@ public ProductDataOut GetProductPricing(Ctx ctx, string fields)
                 }
             }
 
+            /** Price List - ValidFrom date validation ** Dt:01/02/2021 ** Modified By: Kumar **/
+            if (!string.IsNullOrEmpty(Util.GetValueOfString(orderDate)))
+            {
+                StringBuilder sbparams = new StringBuilder();
+                sbparams.Append(Util.GetValueOfInt(M_PriceList_ID));
+                sbparams.Append(",").Append(Util.GetValueOfString(Convert.ToDateTime(orderDate).ToString("MM-dd-yyyy")));
```
Comment (Amit-Vienna) [inline VIS/Areas/VIS/Models/Callouts/MProductPricingModel.cs:65]: not required Util.GetValueOfString

File: ModelLibrary/Model/MProductPricing.cs [hunk 4]
```diff
@@ -205,11 +207,15 @@ private bool CalculatePLV()
                 Tuple<String, String, String> mInfo1 = null;
                 if (Env.HasModulePrefix("ED011_", out mInfo1))
                 {
-                    //vikas  mantis Issue ( 0000517)
-                    string _sql = null;
-                    _sql = "SELECT C_UOM_ID FROM M_Product WHERE  M_Product_ID=" + _M_Product_ID;
-                    _C_UOM_ID = Util.GetValueOfInt(DB.ExecuteScalar(_sql));
-                    //end
+                    /** Price List - Ensuring valid Uom id ** Dt:01/02/2021 ** Modified By: Kumar **/
+                    if (_C_UOM_ID <= UOM_EACH)
```
Comment (Amit-Vienna) [inline ModelLibrary/Model/MProductPricing.cs:211]: no need UOM_EACH variable, when _C_UOM_ID = 0 then  execute this section

**Review by Amit-Vienna — CHANGES_REQUESTED** [review]
change the requested

**Review by Manjot-Vienna — APPROVED** [review]
Now it's OK

**Review by Amit-Vienna — APPROVED** [review]
ok

**Pull request merged**